pkg/azuredisk/nodeserver_v2_test.go

```diff
@@ -30,13 +31,442 @@ import (
+			},
+			expectedErr: testutil.TestError{
+				DefaultError: status.Error(codes.InvalidArgument, "lun not provided"),
+			},
+		},
+		{
+			desc: "[Error] Lun not valid",
+			req: csi.NodePublishVolumeRequest{VolumeCapability: &csi.VolumeCapability{AccessMode: &volumeCap, AccessType: stdVolCapBlock},
+				VolumeId:          "vol_1",
+				TargetPath:        azuredisk,
+				StagingTargetPath: sourceTest,
+				Readonly:          true},
+			setupFunc: func(t *testing.T, d *fakeDriverV2) {
+				d.crdProvisioner.(*mockprovisioner.MockCrdProvisioner).EXPECT().GetAzVolumeAttachment(gomock.Any(), gomock.Any(), gomock.Any()).Return(publishContext, err)
+			},
+			expectedErr: testutil.TestError{
+				DefaultError: status.Error(codes.Internal, "Failed to find device path with lun /dev/01. cannot parse deviceInfo: /dev/01"),
+			},
+		},
+		{
+			desc: "[Error] Mount error mocked by Mount",
+			req: csi.NodePublishVolumeRequest{VolumeCapability: &csi.VolumeCapability{AccessMode: &volumeCap},
+				VolumeId:          "vol_1",
+				TargetPath:        targetTest,
+				StagingTargetPath: errorMountSource,
+				Readonly:          true},
+			skipOnWindows: true, // permission issues
+			expectedErr: testutil.TestError{
+				DefaultError: status.Errorf(codes.Internal, fmt.Sprintf("Could not mount \"%s\" at \"%s\": "+
+					"fake Mount: source error", errorMountSource, targetTest)),
+			},
+		},
+		{
+			desc: "[Success] Valid request already mounted",
+			req: csi.NodePublishVolumeRequest{VolumeCapability: &csi.VolumeCapability{AccessMode: &volumeCap},
+				VolumeId:          "vol_1",
+				TargetPath:        alreadyMountedTarget,
+				StagingTargetPath: sourceTest,
+				Readonly:          true},
+			skipOnWindows: true, // permission issues
+			expectedErr:   testutil.TestError{},
+		},
+		{
+			desc: "[Success] Valid request",
+			req: csi.NodePublishVolumeRequest{VolumeCapability: &csi.VolumeCapability{AccessMode: &volumeCap},
+				VolumeId:          "vol_1",
+				TargetPath:        targetTest,
+				StagingTargetPath: sourceTest,
+				Readonly:          true},
+			skipOnWindows: true, // permission issues
+			expectedErr:   testutil.TestError{},
+		},
+	}
+
+	// Setup
+	_ = makeDir(alreadyMountedTarget)
+	assert.NoError(t, err)
+	fakeMounter, err := mounter.NewFakeSafeMounter()
+	assert.NoError(t, err)
+	d.setMounter(fakeMounter)
+
+	for _, test := range tests {
+		if test.setupFunc != nil {
+			test.setupFunc(t, d)
+		}
+		if !(test.skipOnWindows && runtime.GOOS == "windows") {
+			var err error
+			_, err = d.NodePublishVolume(context.Background(), &test.req)
+			if !testutil.AssertError(&test.expectedErr, err) {
+				t.Errorf("desc: %s\n actualErr: (%v), expectedErr: (%v)", test.desc, err, test.expectedErr.Error())
+			}
+		}
+	}
+
+	// Clean up
+	err = os.RemoveAll(targetTest)
+	assert.NoError(t, err)
+	err = os.RemoveAll(alreadyMountedTarget)
+	assert.NoError(t, err)
+}
+
 func TestNodeStageVolumeMountRecovery(t *testing.T) {
```

Comment:
I believe it makes sense to leave the V2-specific unit tests (i.e. TestNodeStageVolumeMountRecovery & TestRecoverMount) in this file separate from the V1 unit tests.